lib/lightning/runs/query.ex

```diff
@@ -115,14 +126,14 @@ defmodule Lightning.Runs.Query do
   def eligible_for_claim do
     Run
     |> with_cte("in_progress_window", as: ^in_progress_window())
-    |> join(:inner, [r], w in fragment(~s("in_progress_window")),
-      on: r.id == w.id,
+    |> join(:inner, [r], ipw in fragment(~s("in_progress_window")),
+      on: r.id == ipw.id,
       as: :in_progress_window
     )
     |> where(
-      [r, w],
-      (is_nil(w.concurrency) or w.row_number <= w.concurrency) and
-        w.state == "available"
+      [r, ipw],
+      r.state == :available and
+        (is_nil(ipw.concurrency) or ipw.row_number <= ipw.concurrency)
```

Comment:
Super interesting, does this give us a small lift in query performance?

Reply:
A small one but repeatedly might save us some cycles by discarding right away the 'claimed' and 'started' from ipw, that were inserted before the 'available' ones.